### PR TITLE
Adding containsAnyOf assertion

### LINKED ISSUE #954: Feature Request: AbstractIterableAssert.containsAny
#### Summary
I couldn't find an easy way to assert that an Iterable contains any of a given set of values.

#### Example

```java
		// passes:
		assertThat(Arrays.asList("foo")).containsAny("foo", "bar");
		assertThat(Arrays.asList("bar")).containsAny("foo", "bar");
		assertThat(Arrays.asList("foo", "bazz")).containsAny("foo", "bar");
		
		// fails:
		assertThat(Arrays.asList("bazz")).containsAny("foo", "bar");
```

#### Java 8 specific ?
No
* YES : create Pull Request from the `master` branch
* NO  : create Pull Request from the `2.x` branch

### PR DESCRIPTION
**Pull request by marko-bekhta**
#### Check List:
* Fixes #954
* Unit tests : YES 
* Javadoc with a code example (API only) : NO

Hey, that's my first PR here, so if the approach is correct I can add JavaDocs and anything else if missing. Thanks

### CODE REVIEW

File: src/main/java/org/assertj/core/internal/Iterables.java [hunk 1]
```diff
@@ -917,6 +919,18 @@ public void assertContainsExactly(AssertionInfo info, Iterable<?> actual, Object
                            shouldContainExactly(actual, values, diff.missing, diff.unexpected, comparisonStrategy));
   }
 
+  public void containsAnyOf(AssertionInfo info, Iterable<?> actual, Object[] values) {
```
Comment (joel-costigliola) [inline src/main/java/org/assertj/core/internal/Iterables.java]: rename to `assertContainsAnyOf` for consistency with other `Iterables` method.

File: src/main/java/org/assertj/core/internal/Iterables.java [hunk 2]
```diff
@@ -917,6 +919,18 @@ public void assertContainsExactly(AssertionInfo info, Iterable<?> actual, Object
...
+    checkIsNotNull(values);
+    assertNotNull(info, actual);
```
Comment (joel-costigliola) [inline src/main/java/org/assertj/core/internal/Iterables.java]: I believe these checks can be replaced by `commonCheckThatIterableAssertionSucceeds` that would also handle the empty case.

File: src/main/java/org/assertj/core/internal/Iterables.java [hunk 3]
```diff
@@ -917,6 +919,18 @@ public void assertContainsExactly(AssertionInfo info, Iterable<?> actual, Object
...
+    Set<Object> valuesToSearchFor = newTreeSet(values);
+    for (Object element : actual) {
+      if (valuesToSearchFor.contains(element)) {
```
Comment (joel-costigliola) [inline src/main/java/org/assertj/core/internal/Iterables.java]: if possible  in your IDE, remove curly braces and put `return;` in the same line

File: src/main/java/org/assertj/core/internal/Iterables.java [hunk 4]
```diff
@@ -917,6 +919,18 @@ public void assertContainsExactly(AssertionInfo info, Iterable<?> actual, Object
...
+        return;
+      }
+    }
+    throw failures.failure(info, ShouldContainAnyOf.shouldContainAnyOf(actual, values));
```
Comment (joel-costigliola) [inline src/main/java/org/assertj/core/internal/Iterables.java]: static import `shouldContainAnyOf` for readability

**Comment by joel-costigliola** [comment]
@marko-bekhta thanks for the PR ! 
I have reviewed it and I have a few additional comments that did not fit in the PR review: 

- can you add unit tests for `assertContainsAnyOf` in `Iterables_assertContainsAnyOf_Test.java` as in [Iterables_assertContains_Test.java](https://github.com/joel-costigliola/assertj-core/blob/52aa454c7ac2150741b0e98289dbefe1e0b9653d/src/test/java/org/assertj/core/internal/iterables/Iterables_assertContains_Test.java), please try to cover all the corner cases.
- if you have the time, can you add `containsAnyElementsOf` ? it is the same assertion but it takes an `Iterable` in place of a vararg.

**Comment by marko-bekhta** [comment]
@joel-costigliola thanks for the review. I'll try to address all the things that you've mentioned later today.

**Comment by marko-bekhta** [comment]
Changed the things that were mentioned. In addition, changed parameter names to be consistent with others,  and added JavaDocs, used `Iterables#iterableContains` instead of regular `Set#contains`.

Also found one missing param name in the documentation - so added it as well.

File: src/test/java/org/assertj/core/internal/iterables/Iterables_assertContainsAnyOf_Test.java [hunk 5]
```diff
@@ -0,0 +1,155 @@
...
+  }
+
+  @Test
+  public void should_fail_if_actual_is_null() {
+    thrown.expectAssertionError(actualIsNull());
+    iterables.assertContainsAnyOf(someInfo(), null, array("Yoda"));
+  }
+
+  @Test
+  public void should_fail_if_actual_does_not_contain_any_of_values() {
```
Comment (joel-costigliola) [inline src/test/java/org/assertj/core/internal/iterables/Iterables_assertContainsAnyOf_Test.java]: `should_fail_if_actual_does_not_contain_any_of_the_given_values`

File: src/main/java/org/assertj/core/api/AbstractIterableAssert.java [hunk 6]
```diff
@@ -677,6 +678,64 @@ public SELF usingDefaultElementComparator() {
...
+   * @param iterable the iterable of values that, at least one of which is expected to be in the given {@code Iterable}.
+   * @return {@code this} assertion object.
+   * @throws NullPointerException if the iterable of expected values is {@code null}.
+   * @throws IllegalArgumentException if the iterable of expected values is empty.
+   * @throws AssertionError if the given {@code Iterable} is {@code null}.
+   * @throws AssertionError if the given {@code Iterable} does not contain any of elements from other given {@code Iterable}.
+   */
+  public SELF containsAnyElementsOf(Iterable<ELEMENT> iterable) {
+    iterables.assertContainsAnyElementsOf(info, actual, iterable);
+    return myself;
```
Comment (joel-costigliola) [inline src/main/java/org/assertj/core/api/AbstractIterableAssert.java]: it is simpler to call `containsAnyOf` like this: `return containsAnyOf(toArray(iterable));` then the only test you need for this guys is that it delegates correctly to `iterables.assertContainsAnyOf`.
You can also get rid of `iterables.assertContainsAnyElementsOf` and its corresponding tests.

Comment (marko-bekhta) [inline src/main/java/org/assertj/core/api/AbstractIterableAssert.java]: addressed this and updated the commit.

**Comment by joel-costigliola** [comment]
Integrated thank you @marko-bekhta  !
